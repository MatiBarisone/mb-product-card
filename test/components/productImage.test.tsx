import React from "react";
import { render } from "@testing-library/react";
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from './data/products';


describe('ProductImage', () => {

    test('debe de mostrar el componente con la imagen personalizada', () => {
        const { asFragment } = render(
            <ProductCard product={product2}>
                {() => <ProductImage img="https://hola.jpg" />}
            </ProductCard>
        );
    
        expect(asFragment()).toMatchSnapshot();
    });

    test('debe de mostrar el componente con su imagen original', () => {
        const { asFragment } = render(
            <ProductCard product={product2}>
                {() => <ProductImage />}
            </ProductCard>
        );
    
        expect(asFragment()).toMatchSnapshot();
    });

});