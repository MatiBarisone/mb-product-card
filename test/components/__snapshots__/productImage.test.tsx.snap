// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen personalizada 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <img
      alt="Product"
      class="productImg undefined"
      src="https://hola.jpg"
    />
  </div>
</DocumentFragment>
`;

exports[`ProductImage debe de mostrar el componente con su imagen original 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <img
      alt="Product"
      class="productImg undefined"
      src="./coffee-mug.png"
    />
  </div>
</DocumentFragment>
`;
